velox/type/Variant.h

```diff
@@ -396,11 +396,12 @@ class variant {
     }
   }
 
-  std::string toJson(const TypePtr& type = nullptr) const;
+  std::string toJson(const TypePtr& type) const;
 
   /// Used by python binding, do not change signature.
   std::string pyToJson() const {
-    return toJson();
+    const auto type = inferType();
```

Comment:
This code is problematic. inferType() doesn't work and should not be used. For example, it never returns a correct decimal type.

Reply:
I agree. The usage inside `friend std::ostream& operator<<(std::ostream& stream, const variant& k)` will be removed as part of #6215 
The origin of pyToJson() API is interesting. The history says it was added in this commit
https://github.com/facebookincubator/velox/commit/73d2f935b55f084d30557c7be94b9768efb8e56f

But the PR author who is me, did not add it.
https://github.com/facebookincubator/velox/pull/4434/files#diff-4cd72c1c0cce8b5b0e463c07814d664f551e4364e39d32c176f09103ac357144

@mbasmanova, @kgpai can we remove this API?

Reply:
There are no uses of `pyToJson` inside Velox as well.

Reply:
I see one internal usage. CC: @kgpai

Reply:
@kgpai I moved this to change to a separate PR https://github.com/facebookincubator/velox/pull/6472

Reply:
@mbasmanova 
Unfortunately we cant remove the pyToJson function as it is used by Koski. It is binded to python's/koski's 'to_json' which is used quite extensively in python code. 

I will have a look at the your PR @majetideepak . 

Also sometimes we have to make some minor changes to PRs to make them work well with internal dependencies - sometimes these can be legitimate problems with the PRs (like *san failures etc)  , sometimes it might mean adding some code like above to make things work well.

Reply:
@mbasmanova I had an offline discussion with @kgpai. The Python use cases do not require logical types. He is fine with this fix.
@Krishna-Prasad-P-V please add a comment explaining why `pyToJson` has not been modified. You can say "Current Python use cases do not require logical types. Hence, we infer the Type from the variant instead of changing the API."

Reply:
Done.